great_expectations/profile/user_configurable_profiler.py

```diff
@@ -335,10 +350,15 @@ def _validate_semantic_types_dict(self, profile_dataset):
                 "Entries in semantic type dict must be lists of column names e.g. "
                 "{'semantic_types': {'numeric': ['number_of_transactions']}}"
             )
-            if k.upper() not in profiler_semantic_types:
+            if not any(
+                [
+                    k.upper() == semantic_type.value
+                    for semantic_type in ProfilerSemanticTypes
+                ]
+            ):
```

Comment:
So incredibly minor but this could be a generator instead of a list.
```suggestion
            if not any(
                k.upper() == semantic_type.value
                for semantic_type in ProfilerSemanticTypes
            ):
```
Please only change if you're making other change!

Reply:
@cdkini Good idea.